cpp/include/raft/core/managed_mdarray.hpp

```diff
@@ -0,0 +1,178 @@
+using managed_mdarray =
+  mdarray<ElementType, Extents, LayoutPolicy, managed_accessor<ContainerPolicy>>;
+
+/**
+ * @brief Shorthand for 0-dim host mdarray (scalar).
+ * @tparam ElementType the data type of the scalar element
+ * @tparam IndexType the index type of the extents
+ */
+template <typename ElementType, typename IndexType = std::uint32_t>
+using managed_scalar = managed_mdarray<ElementType, scalar_extent<IndexType>>;
+
+/**
+ * @brief Shorthand for 1-dim managed mdarray.
+ * @tparam ElementType the data type of the vector elements
+ * @tparam IndexType the index type of the extents
+ * @tparam LayoutPolicy policy for strides and layout ordering
+ */
+template <typename ElementType,
+          typename IndexType    = std::uint32_t,
+          typename LayoutPolicy = layout_c_contiguous>
+using managed_vector = managed_mdarray<ElementType, vector_extent<IndexType>, LayoutPolicy>;
+
+/**
+ * @brief Shorthand for c-contiguous managed matrix.
+ * @tparam ElementType the data type of the matrix elements
+ * @tparam IndexType the index type of the extents
+ * @tparam LayoutPolicy policy for strides and layout ordering
+ */
+template <typename ElementType,
+          typename IndexType    = std::uint32_t,
+          typename LayoutPolicy = layout_c_contiguous>
+using managed_matrix = managed_mdarray<ElementType, matrix_extent<IndexType>, LayoutPolicy>;
+
+/**
+ * @brief Create a managed mdarray.
+ * @tparam ElementType the data type of the matrix elements
+ * @tparam IndexType the index type of the extents
+ * @tparam LayoutPolicy policy for strides and layout ordering
+ * @param handle raft::resources
+ * @param exts dimensionality of the array (series of integers)
+ * @return raft::managed_mdarray
+ */
+template <typename ElementType,
+          typename IndexType    = std::uint32_t,
+          typename LayoutPolicy = layout_c_contiguous,
+          size_t... Extents>
+auto make_managed_mdarray(raft::resources const& handle, extents<IndexType, Extents...> exts)
+{
+  using mdarray_t = managed_mdarray<ElementType, decltype(exts), LayoutPolicy>;
+
+  typename mdarray_t::mapping_type layout{exts};
+  typename mdarray_t::container_policy_type policy{};
+
+  return mdarray_t{handle, layout, policy};
+}
+
+/**
+ * @brief Create a managed mdarray.
+ * @tparam ElementType the data type of the matrix elements
+ * @tparam IndexType the index type of the extents
+ * @tparam LayoutPolicy policy for strides and layout ordering
+ * @param handle raft::resources
+ * @param mr rmm memory resource used for allocating the memory for the array
+ * @param exts dimensionality of the array (series of integers)
+ * @return raft::managed_mdarray
+ */
+template <typename ElementType,
+          typename IndexType    = std::uint32_t,
+          typename LayoutPolicy = layout_c_contiguous,
+          size_t... Extents>
+auto make_managed_mdarray(raft::resources const& handle,
+                          rmm::mr::managed_memory_resource* mr,
```

Comment:
This overload shouldn't exist at all. I've removed it. If someone needs a more specialized allocator, they should handle that in the `container_policy`, since there is no way to introspect the memory resource and confirm that it's returning managed memory.